src/vcpkg/vcpkgpaths.cpp

```diff
@@ -356,7 +377,26 @@ namespace vcpkg
             m_pimpl->m_manifest_path = manifest_root_dir / "vcpkg.json";
         }
 
-        auto config_file = load_configuration(filesystem, args, root, manifest_root_dir);
+        auto configuration_from_manifest = [&]() -> vcpkg::Optional<Json::Object> {
+            if (auto manifest = m_pimpl->m_manifest_doc.get())
+            {
+                auto manifest_obj = manifest->first;
+                if (auto config_obj = manifest_obj.get("x-vcpkg-configuration"))
+                {
+                    if (!config_obj->is_object())
+                    {
+                        print2(Color::error,
+                               "Failed to parse ",
+                               m_pimpl->m_manifest_path,
+                               ": x-vcpkg-configuration must be an object\n");
+                        Checks::exit_fail(VCPKG_LINE_INFO);
+                    }
+                    return make_optional(config_obj->object());
+                }
+            }
+            return nullopt;
+        }();
```

Comment:
```suggestion
        vcpkg::Optional<Json::Object> configuration_from_manifest;
        if (auto manifest = m_pimpl->m_manifest_doc.get())
        {
            auto manifest_obj = manifest->first;
            if (auto config_obj = manifest_obj.get("x-vcpkg-configuration"))
            {
                if (!config_obj->is_object())
                {
                    print2(Color::error,
                            "Failed to parse ",
                            m_pimpl->m_manifest_path,
                            ": x-vcpkg-configuration must be an object\n");
                    Checks::exit_fail(VCPKG_LINE_INFO);
                }

                configuration_from_manifest = make_optional(config_obj->object());
            }
        }
```

It doesn't seem creating an unnamed lambda just to run the lambda doesn't buy us anything here?

As an aside, our optional really needs an emplace!

Reply:
(This does look like something extractable as a function which might be helpful although I haven't dug into which locals are used)